File: android/src/main/java/com/rivereactnative/RNAlignment.kt
package com.rivereactnative

import app.rive.runtime.kotlin.core.Alignment

enum class RNAlignment(private val mValue: String) {
  TopLeft("topLeft"),
  TopCenter("topCenter"),
  TopRight("topRight"),
  CenterLeft("centerLeft"),
  Center("center"),
  CenterRight("centerRight"),
  BottomLeft("bottomLeft"),
  BottomCenter("bottomCenter"),
  BottomRight("bottomRight");

  override fun toString(): String {
    return mValue
  }

  companion object {

    fun mapToRNAlignment(alignment: String): RNAlignment {
      return valueOf(values().first() { it.toString() == alignment }.name)
    }

    fun mapToRiveAlignment(v: RNAlignment): Alignment {
      return when (v) {
        TopLeft -> Alignment.TOP_LEFT
        TopCenter -> Alignment.TOP_CENTER
        TopRight -> Alignment.TOP_RIGHT
        CenterLeft -> Alignment.CENTER_LEFT
        Center -> Alignment.CENTER
        CenterRight -> Alignment.CENTER_RIGHT
        BottomLeft -> Alignment.BOTTOM_LEFT
        BottomCenter -> Alignment.BOTTOM_CENTER
        BottomRight -> Alignment.BOTTOM_RIGHT
        else -> throw IllegalStateException("Unsupported Alignment type")
      }
    }
  }
}

File: ios/RNAlignment.swift
import Foundation
import RiveRuntime

enum RNAlignment: String {
    case TopLeft = "topLeft"
    case TopCenter = "topCenter"
    case TopRight = "topRight"
    case CenterLeft = "centerLeft"
    case Center = "center"
    case CenterRight = "centerRight"
    case BottomLeft = "bottomLeft"
    case BottomCenter = "bottomCenter"
    case BottomRight = "bottomRight"

    static func mapToRNAlignment(value: String) -> RNAlignment {
        if let rnEnum = RNAlignment(rawValue: value) {
            return rnEnum
        } else {
            // Return a default value instead of crashing
            RCTLogWarn("Unsupported alignment type: \(value), defaulting to Center")
            return .Center
        }
    }

    static func mapToRiveAlignment(rnAlignment: RNAlignment) ->  RiveAlignment {
        switch rnAlignment {
        case .TopLeft:
            return RiveAlignment.topLeft
        case .TopCenter:
            return RiveAlignment.topCenter
        case .TopRight:
            return RiveAlignment.topRight
        case .CenterLeft:
            return RiveAlignment.centerLeft
        case .Center:
            return RiveAlignment.center
        case .CenterRight:
            return RiveAlignment.centerRight
        case .BottomLeft:
            return RiveAlignment.bottomLeft
        case .BottomCenter:
            return RiveAlignment.bottomCenter
        case .BottomRight:
            return RiveAlignment.bottomRight
        }
    }
}

